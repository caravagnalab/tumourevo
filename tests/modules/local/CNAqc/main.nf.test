nextflow_process {

    name "Test Process CNAQC"
    script "modules/local/CNAqc/main.nf"
    process "CNAQC"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                meta = [
                    id : "idProva",
                    tumour_sample: "Set6_42",
                    cancer_type: "PANCANCER" 
                ]
                
                // From cna2cnaqc
                cna_rds = file("${projectDir}/results/formatter/cna2cnaqc/MSeq/Set06/Set6_42/MSeq_Set06_Set6_42_cna.rds", checkIfExists: true)
                // From vcf2cnaqc
                snv_rds = file("${projectDir}/results/formatter/vcf2cnaqc/MSeq/Set06/Set6_42/MSeq_Set06_Set6_42_snv.rds", checkIfExists: true)
                
                // Create a tuple with metadata and file path
                input[0] = tuple(meta, cna_rds)
                input[1] = tuple(meta, snv_rds)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
