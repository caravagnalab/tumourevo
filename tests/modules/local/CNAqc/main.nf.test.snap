{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "idProva",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER"
                        },
                        "idProva_qc.rds:md5,ba87717779c4517929b630128f33df1e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "idProva",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER"
                        },
                        "idProva_data_plot.rds:md5,369fd3d42ba038e9c20a663e9b8af980",
                        "idProva_qc_plot.rds:md5,f8ce8df8b511d4a5cab603ad1a868883"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "idProva",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER"
                        },
                        "idProva_data.pdf:md5,6811f5710ccb50183a6cda5b2384d10d"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "idProva",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER"
                        },
                        "idProva_qc.pdf:md5,a79abb6509a88b0176fce049201f50cd"
                    ]
                ],
                "plot_pdf_data": [
                    [
                        {
                            "id": "idProva",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER"
                        },
                        "idProva_data.pdf:md5,6811f5710ccb50183a6cda5b2384d10d"
                    ]
                ],
                "plot_pdf_qc": [
                    [
                        {
                            "id": "idProva",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER"
                        },
                        "idProva_qc.pdf:md5,a79abb6509a88b0176fce049201f50cd"
                    ]
                ],
                "plot_rds": [
                    [
                        {
                            "id": "idProva",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER"
                        },
                        "idProva_data_plot.rds:md5,369fd3d42ba038e9c20a663e9b8af980",
                        "idProva_qc_plot.rds:md5,f8ce8df8b511d4a5cab603ad1a868883"
                    ]
                ],
                "qc_rds": [
                    [
                        {
                            "id": "idProva",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER"
                        },
                        "idProva_qc.rds:md5,ba87717779c4517929b630128f33df1e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-07-23T12:26:09.4748766"
    }
}